src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,73 @@
+package lotto.domain;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Lotto {
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        checkDuplicate(numbers);
+        this.numbers = numbers;
+    }
+
+    private void validate(List<Integer> numbers) {
+        if (numbers.size() != 6) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    private boolean isDuplicate(int currentIndex, List<Integer> numbers) {
+        Integer currentNumber = numbers.get(currentIndex);
+
+        for (int index = currentIndex + 1; index < numbers.size(); index++) {
+            if (currentNumber.equals(numbers.get(index))) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private void checkDuplicate(List<Integer> numbers) {
+        for (int index = 0; index < numbers.size() - 1; index++) {
+            if (isDuplicate(index, numbers)) {
+                throw new IllegalArgumentException();
+            }
+        }
+    }
```

Comment:
중복을 체크하는 부분은 `Set`이나 `Stream`의 사용으로 훨씬 간단하게 코드를 줄일 수 있으니 한번 학습해 보시면 좋을 것 같아요!!
아래는 `Set`의 사용 예시 입니다.
```suggestion
    private void checkDuplicate(List<Integer> numbers) {
        Set<Integer> set = new HashSet<>(numbers);
        if(set.size() != numbers.size()){
            throw new IllegalArgumentException();
        }
    }
```

Reply:
아하.. set을 깜빡하고 있었네요
사실 저는 이번에 다른 분들 로또 프로젝트 코드리뷰 하면서 stream을 보게 되었어요
처음엔 복잡하다고 생각했는데, 위의 상황에서 stream을 사용하면 함수도 하나로 충분했고 코드 양을 반으로 줄일 수 있었어요
전체 값을 모두 확인해야 할 때는 stream이 훨씬 직관적이어서 좋았던 것 같습니다
set도 적절히 활용하면 중복값을 체크할 때 유용하게 사용할 수 있을 것 같아요ㅎㅎ 감사합니다!

Reply:
맞아요!
모든 순간에서 `Stream`이 좋다 라고는 할 수 없지만 일련의 과정들을 `Stream` 체인으로 연결시키고, 각각의 내부 동작을 코드가 아닌 특정 메서드를 사용하면서 가독성을 높일 수 있다는 것이 `Stream`의 장점인 것 같아요!